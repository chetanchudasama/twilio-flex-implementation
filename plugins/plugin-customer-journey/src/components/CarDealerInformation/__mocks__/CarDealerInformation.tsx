import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
