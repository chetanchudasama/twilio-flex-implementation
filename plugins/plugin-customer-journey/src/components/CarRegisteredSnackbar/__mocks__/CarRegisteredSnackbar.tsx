import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
