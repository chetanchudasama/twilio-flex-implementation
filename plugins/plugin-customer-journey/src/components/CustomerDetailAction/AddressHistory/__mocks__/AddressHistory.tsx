import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
