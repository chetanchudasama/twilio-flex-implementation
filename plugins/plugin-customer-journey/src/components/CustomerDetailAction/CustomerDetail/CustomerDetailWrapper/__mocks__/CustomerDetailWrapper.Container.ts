import CustomerDetailWrapper from "./CustomerDetailWrapper";

export default CustomerDetailWrapper;
