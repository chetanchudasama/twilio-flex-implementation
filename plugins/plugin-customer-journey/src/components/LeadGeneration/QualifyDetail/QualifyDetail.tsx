import React from "react";
import { Divider, Grid, Typography } from "@material-ui/core";
import { CustomSnackbar, Shared, ThreeStateToggle } from "@common/components";
import { QualifyDetailModel } from "../../../models/QualifyDetailModel";
import { QualifyDetailStyles } from "./QualifyDetail.Styles";
import { QualifyDetailProps } from "./QualifyDetail.Props";

export const QualifyDetail: React.FC<QualifyDetailProps> = ({
  monthlyIncome,
  qualifyingDetail,
  updateQualifyingDetails,
}) => {
  const onThreeWaySwitchChange =
    (key: keyof QualifyDetailModel) => (value: boolean | null) => {
      updateQualifyingDetails(key, value);
    };

  return (
    <QualifyDetailStyles>
      <Grid container spacing={16}>
        <Grid item xs={12} className="qualify-detail-heading">
          Qualify
        </Grid>
        <Grid item xs={12}>
          <Grid
            container
            direction="row"
            justify="space-between"
            alignItems="center"
            spacing={8}
          >
            <Grid item>
              <Typography component="p" variant="caption">
                Does the customer have a valid full UK driving license?
              </Typography>
            </Grid>
            <Grid item>
              <ThreeStateToggle
                checked={qualifyingDetail.drivingLicenseConfirmed}
                onChange={onThreeWaySwitchChange("drivingLicenseConfirmed")}
              />
            </Grid>
            {process.env.REACT_APP_AGENT_GUIDE_FOR_LICENSE &&
              qualifyingDetail.drivingLicenseConfirmed === false && (
                <Grid item xs={12}>
                  <CustomSnackbar
                    message={process.env.REACT_APP_AGENT_GUIDE_FOR_LICENSE}
                    type="default"
                  />
                </Grid>
              )}
            <Grid item xs={12}>
              <Divider />
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <Grid
            container
            direction="row"
            justify="space-between"
            alignItems="center"
            spacing={8}
          >
            <Grid item>
              <Typography component="p" variant="caption">
                Is the address on the customer&apos;s driving licence on this
                application?
              </Typography>
            </Grid>
            <Grid item>
              <ThreeStateToggle
                checked={qualifyingDetail.addressConfirmed}
                onChange={onThreeWaySwitchChange("addressConfirmed")}
              />
            </Grid>
            {process.env.REACT_APP_AGENT_GUIDE_FOR_ADDRESS &&
              qualifyingDetail.addressConfirmed === false && (
                <Grid item xs={12}>
                  <CustomSnackbar
                    message={process.env.REACT_APP_AGENT_GUIDE_FOR_ADDRESS}
                    type="default"
                  />
                </Grid>
              )}
            <Grid item xs={12}>
              <Divider />
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <Grid
            container
            direction="row"
            justify="space-between"
            alignItems="center"
            spacing={8}
          >
            <Grid item>
              <Typography
                component="p"
                variant="caption"
                className="income-text"
              >
                {`Is the customer's monthly net income ${Shared.getFormattedCurrencyValue(
                  monthlyIncome
                )}?`}
              </Typography>
            </Grid>
            <Grid item>
              <ThreeStateToggle
                checked={qualifyingDetail.incomeConfirmed}
                onChange={onThreeWaySwitchChange("incomeConfirmed")}
              />
            </Grid>
            {process.env.REACT_APP_AGENT_GUIDE_FOR_INCOME &&
              qualifyingDetail.incomeConfirmed === false && (
                <Grid item xs={12}>
                  <CustomSnackbar
                    message={process.env.REACT_APP_AGENT_GUIDE_FOR_INCOME}
                    type="default"
                  />
                </Grid>
              )}
          </Grid>
        </Grid>
      </Grid>
    </QualifyDetailStyles>
  );
};
