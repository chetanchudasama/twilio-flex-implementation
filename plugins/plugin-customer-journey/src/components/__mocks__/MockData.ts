import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
