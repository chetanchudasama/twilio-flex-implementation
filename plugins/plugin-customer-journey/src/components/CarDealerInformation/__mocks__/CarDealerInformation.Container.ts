import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
