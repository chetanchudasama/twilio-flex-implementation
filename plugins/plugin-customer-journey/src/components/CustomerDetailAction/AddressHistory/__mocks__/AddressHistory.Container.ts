import AddressHistory from "./AddressHistory";

export default AddressHistory;
