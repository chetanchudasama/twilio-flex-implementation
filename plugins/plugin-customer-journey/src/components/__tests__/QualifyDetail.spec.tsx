import { mount } from "enzyme";
import flushPromises from "flush-promises";
import React from "react";
import { create } from "react-test-renderer";

import { createMuiTheme, MuiThemeProvider } from "@material-ui/core";
import { Manager } from "@twilio/flex-ui";

import { Shared } from "@common/components";
import { QualifyDetailModel } from "../../models/QualifyDetailModel";
import { flexManagerMock } from "../__mocks__/MockData";
import { QualifyDetail } from "../LeadGeneration/QualifyDetail/QualifyDetail";
import { QualifyDetailProps } from "../LeadGeneration/QualifyDetail/QualifyDetail.Props";

const defaultProps: QualifyDetailProps = {
  monthlyIncome: 0,
  qualifyingDetail: new QualifyDetailModel(),
  updateQualifyingDetails: jest.fn(),
};

let props = { ...defaultProps };

jest.mock("../../services/application.service");

Manager.getInstance = flexManagerMock;

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    ThreeStateToggle: jest.fn(({ children }) => (
      <div className="Three-State-Toggle-Button">{children}</div>
    )),
  };
});

jest.mock("../../Notifications", () => {
  return {
    CustomNotificationType: { SuccessNotification: "successNotification" },
    showMessage: jest.fn(() => {}),
  };
});

jest.mock("../../services/application.service");

const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
});

describe("QualifyDetail", () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
    props = { ...defaultProps };
  });

  afterAll(() => {
    process.env = OLD_ENV;
  });

  it("renders correctly", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <QualifyDetail {...props} />
      </MuiThemeProvider>
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render agent guide for license message, if drivingLicenseConfirmed property is false", async () => {
    props.qualifyingDetail.drivingLicenseConfirmed = false;
    process.env.REACT_APP_AGENT_GUIDE_FOR_LICENSE =
      "REACT_APP_AGENT_GUIDE_FOR_LICENSE message";
    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <QualifyDetail {...props} />
      </MuiThemeProvider>
    );
    wrapper.update();
    await flushPromises();

    expect(wrapper.html()).toContain(
      "REACT_APP_AGENT_GUIDE_FOR_LICENSE message"
    );
  });

  it("should render agent guide for address message, if addressConfirmed property is false", async () => {
    props.qualifyingDetail.addressConfirmed = false;
    process.env.REACT_APP_AGENT_GUIDE_FOR_ADDRESS =
      "REACT_APP_AGENT_GUIDE_FOR_ADDRESS message";
    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <QualifyDetail {...props} />
      </MuiThemeProvider>
    );
    wrapper.update();
    await flushPromises();

    expect(wrapper.html()).toContain(
      "REACT_APP_AGENT_GUIDE_FOR_ADDRESS message"
    );
  });

  it("should render agent guide for income message, if incomeConfirmed property is false", async () => {
    props.qualifyingDetail.incomeConfirmed = false;
    process.env.REACT_APP_AGENT_GUIDE_FOR_INCOME =
      "REACT_APP_AGENT_GUIDE_FOR_INCOME message";
    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <QualifyDetail {...props} />
      </MuiThemeProvider>
    );
    wrapper.update();
    await flushPromises();

    expect(wrapper.html()).toContain(
      "REACT_APP_AGENT_GUIDE_FOR_INCOME message"
    );
  });

  it("should render customer's monthly income correctly in question text", async () => {
    props.monthlyIncome = 3000;
    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <QualifyDetail {...props} />
      </MuiThemeProvider>
    );

    expect(wrapper.find(".income-text").last().text()).toContain(
      `Is the customer's monthly net income ${Shared.getFormattedCurrencyValue(
        props.monthlyIncome
      )}?`
    );
  });
});
