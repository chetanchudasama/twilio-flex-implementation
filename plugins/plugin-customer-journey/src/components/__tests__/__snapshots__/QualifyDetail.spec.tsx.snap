// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QualifyDetail renders correctly 1`] = `
<div
  className="css-dgid8x"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 qualify-detail-heading"
    >
      Qualify
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108"
          >
            Does the customer have a valid full UK driving license?
          </p>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <div
            className="Three-State-Toggle-Button"
          />
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <hr
            className="MuiDivider-root-134"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108"
          >
            Is the address on the customer's driving licence on this application?
          </p>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <div
            className="Three-State-Toggle-Button"
          />
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <hr
            className="MuiDivider-root-134"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108 income-text"
          >
            Is the customer's monthly net income Â£0?
          </p>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <div
            className="Three-State-Toggle-Button"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
